src/services/ghost/GhostDocumentStore.ts

```diff
@@ -0,0 +1,24 @@
+import * as vscode from "vscode"
+import { GhostDocumentStoreItem } from "./types"
+
+export class GhostDocumentStore {
+	private historyLimit: number = 100 // Limit the number of snapshots to keep
+	private documentStore: Map<string, GhostDocumentStoreItem> = new Map()
+
+	public storeDocument(document: vscode.TextDocument): void {
+		const uri = document.uri.toString()
+		if (!this.documentStore.has(uri)) {
+			this.documentStore.set(uri, {
+				uri,
+				document,
+				history: [],
+			})
+		}
```

Comment:
Is this just for document versioning? I think we have a checkpoint system we might be able to use?